internal/providers/terraform/aws/elasticache_replication_group.go

```diff
@@ -3,27 +3,57 @@ package aws
 import (
 	"github.com/infracost/infracost/internal/resources/aws"
 	"github.com/infracost/infracost/internal/schema"
-	"github.com/tidwall/gjson"
 )
 
 func getElastiCacheReplicationGroupItem() *schema.RegistryItem {
 	return &schema.RegistryItem{
-		Name:  "aws_elasticache_replication_group",
-		RFunc: NewElastiCacheReplicationGroup,
+		Name:                "aws_elasticache_replication_group",
+		RFunc:               NewElastiCacheReplicationGroup,
+		ReferenceAttributes: []string{"aws_appautoscaling_target.resource_id"},
+		CustomRefIDFunc: func(d *schema.ResourceData) []string {
+			// returns a name that will match the custom format used by aws_appautoscaling_target.resource_id
+			name := d.Get("replication_group_id").String()
+			if name != "" {
+				return []string{"replication-group/" + name}
+			}
+			return nil
+		},
 	}
 }
 func NewElastiCacheReplicationGroup(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	cacheClusters := d.Get("num_cache_clusters").Int()
+	if d.IsEmpty("num_cache_clusters") {
+		// check for deprecated attribute
+		cacheClusters = d.Get("number_cache_clusters").Int()
+	}
```

Comment:
```suggestion
	cacheClusters := d.GetInt64OrDefault("num_cache_clusters", 1)
	if d.IsEmpty("num_cache_clusters") {
		// check for deprecated attribute
		cacheClusters = d.Get("number_cache_clusters").Int()
	}
```
`num_cache_clusters` defaults to `1` if not specified and this won't be picked up by hcl